packages/ui/src/components/pagination/pagination.tsx

```diff
@@ -24,14 +25,16 @@ export function Pagination({
   indeterminate = false,
   className
 }: PaginationProps) {
+  const { t } = useTranslation()
+
   const handleGoToPage = (selectedPage?: number) => (e: React.MouseEvent) => {
     e.preventDefault()
     if (!selectedPage) return
 
     goToPage?.(selectedPage)
   }
 
-  const totalPages = indeterminate || !totalItems || !pageSize ? undefined : Math.ceil(totalItems / pageSize)
+  const totalPages = indeterminate || !totalItems ? undefined : Math.ceil(totalItems / pageSize)
```

Comment:
question - so we usually get total pages for a paginated api along with total no of items...which we can use first before falling back to calculating the total based on items ?

<img width="417" height="182" alt="Image" src="https://github.com/user-attachments/assets/038c28ec-4a7d-46a9-a0e2-d12cc6da7278" />

Reply:
What @rkapoor10 is saying does makes sense to me. We can add a prop for `totalPages` and that can take precedence. 

@rkapoor10 can we do it as separate PR, right next to this PR? So that we can avoid big PRs.

cc: @3em